nalgebra-glm references std with the std feature turned off

It looks like an unconditional reference to `std` was added some time ago. According to blame and my tests, the `std` feature last worked as intended in the 0.6 release.

Here's a minimal repro:
`repro/Cargo.toml`
```
[package]
name = "nalgebra-nostd-repro"
version = "0.1.0"
edition = "2021"

[dependencies]
nalgebra-glm = { version = "0.15", default-features = false }
```


`repro/src/main.rs`
```
#![no_std]
```


```
~/repro $ cargo build
   Compiling nalgebra-glm v0.15.0
error[E0433]: failed to resolve: use of undeclared crate or module `std`
 --> /home/mvanbem/.cargo/registry/src/github.com-1ecc6299db9ec823/nalgebra-glm-0.15.0/src/traits.rs:6:5
  |
6 | use std::cmp::PartialOrd;
  |     ^^^ use of undeclared crate or module `std`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `nalgebra-glm` due to previous error
```

I think this could be resolved by switching to the corresponding types exported from the `core` crate. Note that this minimal repro wouldn't compile even with the bug fixed. But it would fail for normal `no_std` setup reasons like not defining a panic handler, rather than this name resolution error in `nalgebra-glm`. I'm not sure how to write a standalone `no_std` example without bringing in an embedded toolchain.

--- a/.github/workflows/nalgebra-ci-build.yml
+++ b/.github/workflows/nalgebra-ci-build.yml
@@ -106,3 +106,7 @@ jobs:
         run: xargo build --verbose --no-default-features --features alloc --target=x86_64-unknown-linux-gnu;
       - name: build thumbv7em-none-eabihf
         run: xargo build --verbose --no-default-features --target=thumbv7em-none-eabihf;
+      - name: build x86_64-unknown-linux-gnu nalgebra-glm
+        run: xargo build --verbose --no-default-features -p nalgebra-glm --target=x86_64-unknown-linux-gnu;
+      - name: build thumbv7em-none-eabihf nalgebra-glm
+        run: xargo build --verbose --no-default-features -p nalgebra-glm --target=thumbv7em-none-eabihf;

--- a/nalgebra-glm/src/traits.rs
+++ b/nalgebra-glm/src/traits.rs
@@ -1,9 +1,9 @@
 use approx::AbsDiffEq;
 use num::{Bounded, Signed};
 
+use core::cmp::PartialOrd;
 use na::Scalar;
 use simba::scalar::{ClosedAdd, ClosedMul, ClosedSub, RealField};
-use core::cmp::PartialOrd;
 
 /// A number that can either be an integer or a float.
 pub trait Number:

--- a/nalgebra-glm/src/traits.rs
+++ b/nalgebra-glm/src/traits.rs
@@ -3,7 +3,7 @@ use num::{Bounded, Signed};
 
 use na::Scalar;
 use simba::scalar::{ClosedAdd, ClosedMul, ClosedSub, RealField};
-use std::cmp::PartialOrd;
+use core::cmp::PartialOrd;
 
 /// A number that can either be an integer or a float.
 pub trait Number:

